### PR TITLE
[ELY-1932] Support for multiple realms - Distributed Identities

### PR DESCRIPTION
**Pull request by nekdozjam**
https://issues.redhat.com/browse/ELY-1932

### CODE REVIEW

File: auth/server/base/src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java [hunk 1]
```diff
@@ -1728,13 +1728,13 @@ boolean verifyEvidence(final Evidence evidence) throws RealmUnavailableException
                 // no verification possible, no identity found
                 return false;
             }
-            final Principal resolvedPrincipal = realmIdentity.getRealmIdentityPrincipal();
-            if (resolvedPrincipal == null) {
-                // we have to have a principal
+            if (! realmIdentity.verifyEvidence(evidence)) {
```
Comment (Skyllarr) [inline auth/server/base/src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java:1731]: @nekdozjam the changes in this file are not necessary, right? But LGTM

File: auth/realm/base/src/main/java/org/wildfly/security/auth/realm/DistributedSecurityRealm.java [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        }
+
+        @Override
+        public AuthorizationIdentity getAuthorizationIdentity() throws RealmUnavailableException {
+            return currentIdentity.getAuthorizationIdentity();
+        }
+
+        @Override
+        public void dispose() {
+            for (RealmIdentity identity : loadedIdentities) {
```
Comment (fjuma) [inline auth/realm/base/src/main/java/org/wildfly/security/auth/realm/DistributedSecurityRealm.java:261]: It doesn't look like `loadedIdentities` actually gets populated so it looks like this block should be removed and replaced with `currentIdentity.dispose()`.

Comment (fjuma) [reply auth/realm/base/src/main/java/org/wildfly/security/auth/realm/DistributedSecurityRealm.java:261]: Will fix this in a follow-up PR.

Comment (fjuma) [reply auth/realm/base/src/main/java/org/wildfly/security/auth/realm/DistributedSecurityRealm.java:261]: Fixed this in https://github.com/wildfly-security/wildfly-elytron/pull/1426

**Pull request merged**